pom.xml

```diff
@@ -2339,39 +2339,6 @@
           </execution>
         </executions>
       </plugin>
-      <plugin>
-        <groupId>org.jacoco</groupId>
-        <artifactId>jacoco-maven-plugin</artifactId>
-        <version>0.8.12</version>
-        <executions>
-          <execution>
-            <goals>
-              <goal>prepare-agent</goal>
-            </goals>
-          </execution>
-          <execution>
-            <id>report</id>
-            <phase>test</phase>
-            <goals>
-              <goal>report-aggregate</goal>
-            </goals>
-            <configuration>
-              <includeCurrentProject>true</includeCurrentProject>
-            </configuration>
-          </execution>
-        </executions>
-        <configuration>
-          <append>true</append>
-          <includes>
-            <include>org/apache/pinot/**/*</include>
-          </includes>
-        </configuration>
-      </plugin>
-      <plugin>
-        <groupId>org.codehaus.mojo</groupId>
-        <artifactId>sonar-maven-plugin</artifactId>
-        <version>2.7.1</version>
-      </plugin>
```

Comment:
+1 to gonzalo's suggestion. Jacoco generates a lot of HTML files and reports that are really not needed. It only runs during the test phase so most of the time doesn't hamper local development as well

Reply:
Codecov github actions supports a list of files to upload but we don't use that because by default codecov looks for all jacoco.xml files in the file tree. You can see that in action in https://github.com/apache/pinot/actions/runs/10482523110/job/29033779007?pr=13859 clicking in `Upload coverage to codecov`

Reply:
Can we also specify sonar plugin in coverage profile? That way we can be sure the builds from GitHub Actions keep producing the same artifacts.

Reply:
AFAIK the sonar-plugin is only used to communicate with SonarQube, but if it does more, the plugin should be moved to the codecoverage profile

Reply:
@xiangfu0 @jfim Do you think we can remove the sonar plugin?